controller/eks-cluster-config-handler.go

```diff
@@ -1171,6 +1175,16 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 		}
 	}
 
+	// once the cluster is created, update the node group launch template with the default security group
+	// generated by EKS
+	addDefaultSecurityGroup := false
+	if !reflect.DeepEqual(config.Status.SecurityGroups, upstreamSpec.SecurityGroups) {
+		logrus.Infof("adding default security group to node groups for cluster [%s]", config.Name)
+		addDefaultSecurityGroup = true
+		config = config.DeepCopy()
+		config.Status.SecurityGroups = upstreamSpec.SecurityGroups
```

Comment:
I think whenever we do a change to config we want to update it with the client because we don't want to end up returning it from the handler if it's not the true state of the object and the client will give us the true state whether there's an error or not. See here for example: https://github.com/rancher/eks-operator/pull/68/files#diff-a74e30a65934de22a0dafc6fd3a31ff5c5c81d47e8fbd43f4f740c419977b565R1125.